test.js

```diff
@@ -56,3 +56,99 @@ it('should calculate minimal base for 2391484', () => {
 it('should calculate minimal base for 1000000000000000000', () => {
   assert.equal(getBase('1000000000000000000'), '999999999999999999');
 });
+let overload = Number.MAX_SAFE_INTEGER;
+
+describe("Manual tests", function(){
+  it ("test base < 10", function(){
+    assert.equal(getBase(3), 2);
+    assert.equal(getBase(7), 2);
+    assert.equal(getBase(21), 4);
+    assert.equal(getBase(57), 7);
+    assert.equal(getBase(1111), 10);
+  });
+
+  it ("big base tests", function(){
+    assert.equal(getBase(1000002), 1000001);
+    assert.equal(getBase(1000000002), 1000000001);
+    assert.equal(getBase(1000000000000), 999999999999);
+  });
+
+  it ("tricky part : )", function(){
+    assert.equal(getBase(1001001), 1000);
+    assert.equal(getBase(1001001001), 1000);
+    assert.equal(getBase(1001001001001), 1000);
+
+    assert.equal(getBase(2500050001), 50000);
+    assert.equal(getBase(125002500050001), 50000);
+  });
+
+
+});
+
+describe("Random tests", function () {
+  it ("random test with base < 10", function(){
+    for (let i=0; i<100; i++) {
+      let n = 2 + ~~(Math.random()*8);
+      let num = 1 + n + n*n + n*n*n + n*n*n*n;
+      assert.equal(getBase(num), n);
+    }
+  });
+
+  it ("random test with base < 1000", function(){
+    for (let i=0; i<100; i++) {
+      let n = 2 + ~~(Math.random()*998);
```

Comment:
Opened tests should not be random, because this is the main purpose - open principle and easy check what is not correct. There will be `pro` tests on padawans soon and i could add these tests. 
Now you can just generate some numbers and create open-tests manually.